src/redux/sagas/motions/expenditures/finalizeExpenditureMotion.ts

```diff
@@ -0,0 +1,141 @@
+import {
+  ClientType,
+  Id,
+  getPermissionProofs,
+  ColonyRole,
+} from '@colony/colony-js';
+import { call, put, takeEvery } from 'redux-saga/effects';
+
+import { ADDRESS_ZERO, APP_URL } from '~constants/index.ts';
+import { type Action, ActionTypes } from '~redux/index.ts';
+import {
+  createGroupTransaction,
+  createTransactionChannels,
+  waitForTxResult,
+} from '~redux/sagas/transactions/index.ts';
+import {
+  getColonyManager,
+  initiateTransaction,
+  putError,
+  takeFrom,
+} from '~redux/sagas/utils/index.ts';
+
+export type FinalizeExpenditureMotionPayload =
+  Action<ActionTypes.MOTION_EXPENDITURE_FINALIZE>['payload'];
+
+function* finalizeExpenditureMotion({
+  meta,
+  meta: { setTxHash },
+  payload: { colony, votingReputationAddress, expenditure, motionDomainId },
+}: Action<ActionTypes.MOTION_EXPENDITURE_FINALIZE>) {
+  const { createMotion } = yield call(createTransactionChannels, meta.id, [
+    'createMotion',
+  ]);
+
+  try {
+    if (
+      !colony ||
+      !expenditure ||
+      !votingReputationAddress ||
+      !motionDomainId
+    ) {
+      throw new Error('Invalid payload');
+    }
+
+    if (colony.version < 15) {
+      throw new Error(
+        'Motions to finalize expenditure are only available in Colony version 15 and above',
+      );
+    }
+
+    const colonyManager = yield call(getColonyManager);
+    const colonyClient = yield call(
+      [colonyManager, colonyManager.getClient],
+      ClientType.ColonyClient,
+      colony.colonyAddress,
+    );
+
+    const [permissionDomainId, childSkillIndex] = yield getPermissionProofs(
+      colonyClient.networkClient,
+      colonyClient,
+      expenditure.nativeDomainId,
+      ColonyRole.Arbitration,
+      votingReputationAddress,
+    );
+
+    const { skillId } = yield call(
+      [colonyClient, colonyClient.getDomain],
+      Id.RootDomain,
+    );
+
+    const { key, value, branchMask, siblings } = yield call(
+      [colonyClient, colonyClient.getReputation],
+      skillId,
+      ADDRESS_ZERO,
+    );
+
+    const encodedAction = colonyClient.interface.encodeFunctionData(
+      'finalizeExpenditureViaArbitration',
+      [permissionDomainId, childSkillIndex, expenditure.nativeId],
+    );
+
+    const batchKey = 'createMotion';
+
+    yield createGroupTransaction(createMotion, batchKey, meta, {
+      context: ClientType.VotingReputationClient,
+      methodName: 'createMotion',
+      identifier: colony.colonyAddress,
+      params: [
+        motionDomainId,
+        childSkillIndex,
+        ADDRESS_ZERO,
+        encodedAction,
+        key,
+        value,
+        branchMask,
+        siblings,
+      ],
+      group: {
+        key: batchKey,
+        id: meta.id,
+        index: 1,
```

Comment:
Nitpick, and I don't remember this clearly, but I remember something as `index` only being required if you wanted to overwrite it (for custom transaction display)

Otherwise, you can just leave it out and it will progress incrementally